src/mame/drivers/tsconf.cpp

```diff
@@ -0,0 +1,293 @@
+	map(0x00ff, 0x00ff).rw(m_beta, FUNC(beta_disk_device::state_r), FUNC(beta_disk_device::param_w)).mirror(0xff00);
+	map(0x00af, 0x00af).rw(FUNC(tsconf_state::tsconf_port_xxaf_r), FUNC(tsconf_state::tsconf_port_xxaf_w)).select(0xff00);
+	map(0x8ff7, 0x8ff7).w(FUNC(tsconf_state::tsconf_port_f7_cmos_w)).select(0x7000); // 3:bff7 5:dff7 6:eff7
+	map(0xbff7, 0xbff7).r(FUNC(tsconf_state::tsconf_port_f7_cmos_r));
+}
+
+void tsconf_state::tsconf_switch(address_map &map)
+{
+	map(0x0000, 0x3fff).r(FUNC(tsconf_state::beta_neutral_r)); // Overlap with next because we want real addresses on the 3e00-3fff range
+	map(0x3d00, 0x3dff).r(FUNC(tsconf_state::beta_enable_r));
+	map(0x4000, 0xffff).r(FUNC(tsconf_state::beta_disable_r));
+}
+
+template <unsigned Bank>
+void tsconf_state::tsconf_bank_w(offs_t offset, u8 data)
+{
+	tsconf_state::ram_bank_write(Bank, offset, data);
+}
+
+static const gfx_layout spectrum_charlayout =
+{
+	8, 8, /* 8 x 8 characters */
+	96,	  /* 96 characters */
+	1,	  /* 1 bits per pixel */
+	{0},  /* no bitplanes */
+	/* x offsets */
+	{STEP8(0, 1)},
+	/* y offsets */
+	{STEP8(0, 8)},
+	8 * 8 /* every char takes 8 bytes */
+};
+
+static const gfx_layout tsconf_charlayout =
+{
+	8,
+	8,
+	256,
+	1,
+	{0},
+	{STEP8(0, 1)},
+	{STEP8(0, 8)},
+	8 * 8
+};
+
+static const gfx_layout tsconf_tile_16cpp_layout =
+{
+	8,
+	8,
+	64 * 64 * 8,
+	4,
+	{STEP4(0, 1)},
+	{STEP8(0, 4)},
+	{STEP8(0, 256 * 8)},
+	// Much more tiles when needed. Because tiles are in RAW formut but we don't know region properties.
+	8 * 4
+};
+
+static GFXDECODE_START(gfx_tsconf)
+	GFXDECODE_ENTRY("maincpu", 0x1fd00, spectrum_charlayout, 0xf7, 1)
+	GFXDECODE_ENTRY("maincpu", 0, tsconf_charlayout, 0xf7, 1)       // TXT
+	GFXDECODE_ENTRY("maincpu", 0, tsconf_tile_16cpp_layout, 0, 255) // T0 16cpp
+	GFXDECODE_ENTRY("maincpu", 0, tsconf_tile_16cpp_layout, 0, 255) // T1 16cpp
+GFXDECODE_END
+
+void tsconf_state::video_start()
+{
+	spectrum_128_state::video_start();
+
+	m_ts_tilemap[0] = &machine().tilemap().create(*m_gfxdecode, tilemap_get_info_delegate(*this, FUNC(tsconf_state::get_tile_info_txt)), TILEMAP_SCAN_ROWS, 8, 8, 128, 64);
+	m_ts_tilemap[1] = &machine().tilemap().create(*m_gfxdecode, tilemap_get_info_delegate(*this, FUNC(tsconf_state::get_tile_info_16c<0>)), TILEMAP_SCAN_ROWS, 8, 8, 64, 64);
+	m_ts_tilemap[2] = &machine().tilemap().create(*m_gfxdecode, tilemap_get_info_delegate(*this, FUNC(tsconf_state::get_tile_info_16c<1>)), TILEMAP_SCAN_ROWS, 8, 8, 64, 64);
+
+	m_ts_tilemap[1]->set_transparent_pen(0);
+	m_ts_tilemap[2]->set_transparent_pen(0);
+}
+
+void tsconf_state::machine_start()
+{
+	m_glukrs->set_base(m_cmos->pointer(), 0x100);
+
+	save_item(NAME(m_regs));
+}
+
+void tsconf_state::machine_reset()
+{
+	u8 *messram = m_ram->pointer();
+	m_program = &m_maincpu->space(AS_PROGRAM);
+	m_p_rom = memregion("maincpu")->base();
+
+	m_ram_0000 = nullptr;
+
+	rgb_t colors[256] = {rgb15(0)};
+	subdevice<palette_device>("palette")->set_pen_colors(0, colors);
```

Comment:
You should use a palette initialisation function for this.

Reply:
Not sure. Here I expand pallete which used in parent. It needs to be functional in base for zx-mode. As it's already initialized in super() I'm don't know how to do what you asked

Reply:
In your machine configuration function, you can do something like:
```
    subdevice<palette_device>("palette")->set_init(FUNC(...));
```
This will set the palette initialisation function, overriding whatever the base set.